Mage.Sets/src/mage/cards/s/SanctuaryBlade.java

```diff
@@ -0,0 +1,47 @@
+package mage.cards.s;
+
+import mage.abilities.common.AttachedToCreatureTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.effects.common.continuous.GainProtectionFromColorAttachedEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class SanctuaryBlade extends CardImpl {
+
+    public SanctuaryBlade(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{2}");
+        
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // As Sanctuary Blade becomes attached to a creature, choose a color.
+        GainProtectionFromColorAttachedEffect protectionEffect = new GainProtectionFromColorAttachedEffect(Duration.WhileOnBattlefield);
```

Comment:
There are already `GainAbilityAttachedEffect`, `ProtectionChosenColorSourceEffect` and `ChooseColorEffect`. Even `ProtectionChosenColorAttachedEffect`. You must use existed classes as much as possible.

Reply:
Yeah thanks for the pointers, updated it to use `ChooseColorEffect` and `ProtectionChosenColorAttachedEffect` (and got rid of extra class I added that combines those two).